test/server/config_validation/xds_fuzz.cc

```diff
@@ -0,0 +1,244 @@
+  sendDiscoveryResponse<envoy::config::listener::v3::Listener>(Config::TypeUrl::get().Listener,
+                                                               listeners, added_or_updated, removed,
+                                                               std::to_string(version_));
+}
+
+void XdsFuzzTest::updateRoute(
+    const std::vector<envoy::config::route::v3::RouteConfiguration> routes,
+    const std::vector<envoy::config::route::v3::RouteConfiguration>& added_or_updated,
+    const std::vector<std::string>& removed) {
+  ENVOY_LOG_MISC(debug, "Sending Route DiscoveryResponse version {}", version_);
+  sendDiscoveryResponse<envoy::config::route::v3::RouteConfiguration>(
+      Config::TypeUrl::get().RouteConfiguration, routes, added_or_updated, removed,
+      std::to_string(version_));
+}
+
+XdsFuzzTest::XdsFuzzTest(const test::server::config_validation::XdsTestCase& input,
+                         envoy::config::core::v3::ApiVersion api_version)
+    : HttpIntegrationTest(
+          Http::CodecClient::Type::HTTP2,
+          input.config().ip_version() == test::server::config_validation::Config::IPv4
+              ? Network::Address::IpVersion::v4
+              : Network::Address::IpVersion::v6,
+          ConfigHelper::adsBootstrap(input.config().sotw_or_delta() ==
+                                             test::server::config_validation::Config::SOTW
+                                         ? "GRPC"
+                                         : "DELTA_GRPC",
+                                     api_version)),
+      actions_(input.actions()), version_(1), api_version_(api_version) {
+  use_lds_ = false;
+  create_xds_upstream_ = true;
+  tls_xds_upstream_ = false;
+
+  parseConfig(input);
+}
+
+void XdsFuzzTest::parseConfig(const test::server::config_validation::XdsTestCase& input) {
+  if (input.config().ip_version() == test::server::config_validation::Config::IPv4) {
+    ip_version_ = Network::Address::IpVersion::v4;
+  } else {
+    ip_version_ = Network::Address::IpVersion::v6;
+  }
+
+  if (input.config().client_type() == test::server::config_validation::Config::GOOGLE_GRPC) {
+    client_type_ = Grpc::ClientType::GoogleGrpc;
+  } else {
+    client_type_ = Grpc::ClientType::EnvoyGrpc;
+  }
+
+  if (input.config().sotw_or_delta() == test::server::config_validation::Config::SOTW) {
+    sotw_or_delta_ = Grpc::SotwOrDelta::Sotw;
+  } else {
+    sotw_or_delta_ = Grpc::SotwOrDelta::Delta;
+  }
+}
+
+/**
+ * initialize an envoy configured with a fully dynamic bootstrap with ADS over gRPC
+ */
+void XdsFuzzTest::initialize() {
+  config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    auto* ads_config = bootstrap.mutable_dynamic_resources()->mutable_ads_config();
+    auto* grpc_service = ads_config->add_grpc_services();
+
+    std::string cluster_name = "ads_cluster";
+    switch (client_type_) {
```

Comment:
I thin client type isn't super interesting to fuzz either, but it's probably a bit more interesting than IP version.

Reply:
removed